apps/patient/src/api/internal.ts

```diff
@@ -0,0 +1,144 @@
+import { types } from 'packages/sdk/dist/lib';
+
+import { graphQLClient } from '../configs/graphqlClient';
+import {
+  GET_ORDER,
+  GET_PHARMACIES,
+  MARK_ORDER_AS_PICKED_UP,
+  REROUTE_ORDER,
+  SELECT_ORDER_PHARMACY,
+  SET_PREFERRED_PHARMACY
+} from '../graphql';
+import { Order } from '../utils/models';
+
+export const AUTH_HEADER_ERRORS = ['EMPTY_AUTHORIZATION_HEADER', 'INVALID_AUTHORIZATION_HEADER'];
+
+/**
+ * Queries
+ */
+
+export const getOrder = async (orderId: string) => {
+  // Not wrapped in try/catch so error handling can be done in Main
+  const response: { order: Order } = await graphQLClient.request(GET_ORDER, {
+    id: orderId
+  });
+  if (response.order) {
+    return response.order;
+  } else {
+    throw new Error('No order found');
+  }
+};
+
+export const getPharmacies = async (
+  searchParams: {
+    latitude: number;
+    longitude: number;
+    radius: number;
+  },
+  limit: number,
+  offset: number
+) => {
+  try {
+    const response: { pharmaciesByLocation: types.Pharmacy[] } = await graphQLClient.request(
+      GET_PHARMACIES,
+      {
+        location: searchParams,
+        limit,
+        offset
+      }
+    );
+    if (response?.pharmaciesByLocation?.length > 0) {
+      return response.pharmaciesByLocation;
+    } else {
+      throw new Error('No pharmacies found near location');
+    }
+  } catch (error) {
+    if (error?.response?.errors?.[0].message === 'No pharmacies found near location') {
+      throw new Error(error.response.errors[0].message);
+    } else {
+      throw error;
+    }
+  }
+};
+
+/**
+ * Mutations
+ */
+
+export const markOrderAsPickedUp = async (orderId: string) => {
+  try {
+    const response: { markOrderAsPickedUp: boolean } = await graphQLClient.request(
+      MARK_ORDER_AS_PICKED_UP,
+      { markOrderAsPickedUpId: orderId }
```

Comment:
`markOrderAsPickedUpId` is a lil confusing, maybe just `orderId`?

Reply:
yeah I agree, not sure why we originally did that, updating will require updating the services backend so I'll leave for next time i revisit that mutation